File: tasking/sources/tasking.9.c
/*
* @@name:	tasking.9
* @@type:	C
* @@operation:	compile
* @@expect:	rt-error
* @@version:	omp_3.0
*/
void work()
{
   #pragma omp task
   { //Task 1
       #pragma omp task
       { //Task 2
            #pragma omp critical //Critical region 1
            {/*do work here */ }
       }
       #pragma omp critical //Critical Region 2
       {
           //Capture data for the following task
           #pragma omp task
           { /* do work here */ } //Task 3
       }
   }
}

File: tasking/sources/tasking.9.f
! @@name:	tasking.9
! @@type:	F-fixed
! @@operation:	compile
! @@expect:	rt-error
! @@version:	omp_3.0
       module example
       contains
       subroutine work
!$omp task
       ! Task 1
!$omp task
       ! Task 2
!$omp critical
       ! Critical region 1
       ! do work here
!$omp end critical
!$omp end task
!$omp critical
       ! Critical region 2
       ! Capture data for the following task
!$omp task
       !Task 3
       ! do work here
!$omp end task
!$omp end critical
!$omp end task
      end subroutine
      end module
